TelCoColorCoder.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef TELCOCOLORCODER_HPP_
+#define TELCOCOLORCODER_HPP_
+
+namespace TelCoColorCoder {
+enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
+enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
+
+extern const char* MajorColorNames[];
+extern int numberOfMajorColors;
+extern const char* MinorColorNames[];
+
+extern int numberOfMinorColors;
+
+// forward declaration
+class ColorPair;
```

Comment:
Forward declaration is useful in breaking circular dependencies. Here, `TelCoColorCoder.hpp` needs `ColorPair` and `ColorPair.hpp` requires this header. In such situations, ask if a different breakup of files will eliminate the need for circular dependency. E.g., move the enums to `ColorPair.hpp`, and keep only the `Get...` functions in this hpp? That will still preserve the separation of concerns, will it not?